### DISCUSSION TITLE
play dsd file stutter on mpd+upmpdcli in ubuntu 22 or 24 LTS.

### DISCUSSION THREAD

**Discussion by YTSamLee** [MusicPlayerDaemon/MPD#2133] [Q&A]
First of all, please forgive my poor English！
1、output of “mpd --version”
2、my configuration file (mpd.conf)
3、relevant portions of the log file (--verbose)
#be clear about what you expect MPD to do, and what is actually happening

I use mpd+upmpdcli to build upnp renderer on Raspberry Pi or X86 low-power machine. play my audio file (main pcm and dsd format) on upnp server(minidlna install on ubuntu or debain);

i can play pcm and dsd audio Perfectly on debain 12( raspberry pi 4 or x86 intel atom cpu);

but On Ubuntu 22 or 24, whether the stock kernel or other low-latency or real-time kernel, playing DSD is always stuttered after a few seconds。
I want to know, can this stuttering be solved in Ubuntu?how?
[mpd --version.txt](https://github.com/user-attachments/files/17273616/mpd.--version.txt)
[mpd.conf.txt](https://github.com/user-attachments/files/17273643/mpd.conf.txt)
[mpd.log](https://github.com/user-attachments/files/17273617/mpd.log)

===========
Finally, I also attach the information about x86 machine on Debian
[debain version.txt](https://github.com/user-attachments/files/17273663/debain.version.txt)

My USB DAC is Holo Audio Cyan 2。
Looking forward to your recovery！

**Comment by YTSamLee** [comment]
This is the state file of the DSD playback.
===========
sw_volume: 56
audio_device_state:1:Holo Audio Cyan 2
state: stop
random: 0
repeat: 0
single: 0
consume: 0
crossfade: 0
mixrampdb: 0.000000
mixrampdelay: -1.000000
playlist_begin
song_begin: http://192.168.200.250:8200/MediaItems/1544.dsd
Artist: Norah Jones
Album: Feels Like Home (DSD)
Title: Sunrise
Track: 1
song_end
playlist_end

sw_volume: 56
audio_device_state:1:Holo Audio Cyan 2
state: stop
random: 0
repeat: 0
single: 0
consume: 0
crossfade: 0
mixrampdb: 0.000000
mixrampdelay: -1.000000
playlist_begin
song_begin: http://192.168.200.250:8200/MediaItems/245.dsd
Artist: UNAMAS Strings Quartet
Album: 维瓦尔第：四季
Title: Concerto in E major,"La Primavera" (Spring) ,Op. 8-1, RV 269
Track: 1
song_end
playlist_end

**Comment by MaxKellermann** [comment]
```
Oct 07 05:25 : exception: CURL failed: Out of memory
```
This is bad. Fix this first.

**Reply by YTSamLee** [reply]
How do I fix this? It's not easy for me to use alsa+mpd+upmpdcli. Do I need to program it? I'm just confused. Why is there no problem under Debian? Is it possible that I can only use mpd+upmpdcli under Debian?

**Reply by MaxKellermann** [reply]
I don't know, it's a CURL problem, not a MPD problem. This is not the right place to ask about CURL problems.

**Comment by YTSamLee** [comment]
debain 12
curl --version
curl 7.88.1 (x86_64-pc-linux-gnu) libcurl/7.88.1 OpenSSL/3.0.14 zlib/1.2.13 brotli/1.0.9 zstd/1.5.4 libidn2/2.3.3 libpsl/0.21.2 (+libidn2/2.3.3) libssh2/1.10.0 nghttp2/1.52.0 librtmp/2.3 OpenLDAP/2.5.13
Release-Date: 2023-02-20, security patched: 7.88.1-10+deb12u7
Protocols: dict file ftp ftps gopher gophers http https imap imaps ldap ldaps mqtt pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli GSS-API HSTS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM NTLM_WB PSL SPNEGO SSL threadsafe TLS-SRP UnixSockets zstd

**Comment by YTSamLee** [comment]
ubuntu 24 lts
curl --version
curl 8.5.0 (x86_64-pc-linux-gnu) libcurl/8.5.0 OpenSSL/3.0.13 zlib/1.3 brotli/1.1.0 zstd/1.5.5 libidn2/2.3.7 libpsl/0.21.2 (+libidn2/2.3.7) libssh/0.10.6/openssl/zlib nghttp2/1.59.0 librtmp/2.3 OpenLDAP/2.6.7
Release-Date: 2023-12-06, security patched: 8.5.0-2ubuntu10.4
Protocols: dict file ftp ftps gopher gophers http https imap imaps ldap ldaps mqtt pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli GSS-API HSTS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM PSL SPNEGO SSL threadsafe TLS-SRP UnixSockets zstd

**Reply by MaxKellermann** [reply]
The "S" in "LTS" means "Support". Ask for this support that was promised to you. Off-topic here.
